pubsub/gcloud/aio/pubsub/publisher_client.py

```diff
@@ -54,13 +56,17 @@ def topic_path(cls, project: str, topic: str) -> str:
         return f'{cls.project_path(project)}/topics/{topic}'
 
     async def _headers(self) -> Dict[str, str]:
+        headers = {
+            'Content-Type': 'application/json'
+        }
         if PUBSUB_EMULATOR_HOST:
-            return {}
+            return headers
 
         token = await self.token.get()
-        return {
-            'Authorization': f'Bearer {token}',
-        }
+        headers.update({
```

Comment:
nit: overkill, use `headers['Authorization'] = f'Bearer {token}'`

Generally, treat `update` as a slower multitool that you need to pull out only for edge-cases; it lets you do more advanced dictionary merging behavior, but that comes with something like a >5x perf loss. Obviously, dictionary updates are "fast enough" for this to not matter all that much, but it's a good default to have.

Reply:
Thanks for the stat! I just copied this from subscriber so I'll update it there too to keep them consistent.